src/base/database.h

```diff
@@ -201,6 +201,21 @@ class Database {
   void DeleteInlierMatches(const image_t image_id1,
                            const image_t image_id2) const;
 
+  // Clear all database tables
+  void ClearAllTables() const;
+
+  // Clear the entire cameras table
+  void ClearCameras() const;
+
+  // Clear the entire images, keypoints, and descriptors tables
+  void ClearImages() const;
+
+  // Clear the entire descriptors table
+  void ClearDescriptors() const;
+
+  // Clear the entire keypoints table
+  void ClearKeypoints() const;
```

Comment:
Would be good to add basic unit tests for these methods in the database_test.cc file.

Reply:
Added testing for clearing tables as part of the existing unit tests. If that's not desirable I can write individual tests for this functionality.